bee-inx/src/protocol_parameters.rs

```diff
@@ -1,32 +1,32 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use bee_block::protocol::ProtocolParameters;
+use bee_block as bee;
 use inx::proto;
-use packable::PackableExt;
 
-use crate::Error;
+use crate::Raw;
 
 #[allow(missing_docs)]
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct RawProtocolParameters {
+pub struct ProtocolParameters {
     pub protocol_version: u8,
-    pub params: Vec<u8>,
+    pub params: Raw<bee::protocol::ProtocolParameters>,
 }
 
-impl RawProtocolParameters {
-    pub fn inner(self) -> Result<ProtocolParameters, Error> {
-        let unpacked = ProtocolParameters::unpack_verified(self.params, &())
-            .map_err(|e| bee_block::InxError::InvalidRawBytes(format!("{:?}", e)))?;
-        Ok(unpacked)
+impl From<proto::RawProtocolParameters> for ProtocolParameters {
+    fn from(value: proto::RawProtocolParameters) -> Self {
+        Self {
+            protocol_version: value.protocol_version as u8,
+            params: value.params.into(),
+        }
     }
 }
 
-impl From<proto::RawProtocolParameters> for RawProtocolParameters {
-    fn from(value: proto::RawProtocolParameters) -> Self {
+impl From<ProtocolParameters> for proto::RawProtocolParameters {
```

Comment:
Is this not the same as in `src/protocol.rs` ?

Reply:
Unfortunately not this is `impl From<bee_inx::ProtocolParameters> for proto::RawProtocolParameters`, whereas the one in `bee-block/src/protocol.rs` is `impl From<bee_block::protocol::ProtocolParameters> for proto::RawProtocolParameters`. I think both conversions can be valuable.

I'm happy to rename `bee_inx::ProtocolParameters` to something more descriptive, but couldn't come up with a better name yet.